api/app/models/fetch/predictions.py

```diff
@@ -192,6 +190,68 @@ async def _fetch_model_predictions_by_station_codes(model: ModelEnum, station_co
 
 
 async def fetch_model_predictions(model: ModelEnum, station_codes: List[int]):
-    """ Fetch 10 day global model weather predictions for a given station."""
+    """ Fetch model weather predictions for a given list of stations and a given model. """
     # Fetch predictions from the database.
     return await _fetch_model_predictions_by_station_codes(model, station_codes)
+
+
+async def _fetch_most_recent_historic_predictions_by_station_codes(model: ModelEnum, station_codes: List[int]) -> List[WeatherModelPrediction]:
+    """ Fetch the most recent historic model predictions from database based on each station's coordinates. """
+    # construct temporary helper dictionary of relevant weather stations
+    stations_dict = {}
+    stations = await app.stations.get_stations_by_codes(station_codes)
+    for station in stations:
+        stations_dict[station.code] = station
+
+    # construct helper dictionary of WeatherModelPredictions
+    # weather_model_predictions_dict is indexed by station_code, then by prediction_timestamp
+    weather_model_predictions_dict = defaultdict(dict)
+
+    # We're only interested in the last 5 days
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
```

Comment:
Please use the one in time_utils - makes for easier deterministic unit test writing.
```suggestion
    now = app.time_utils.get_utc_now()
```